src/compile.rs

```diff
@@ -682,6 +706,145 @@ fn check_required_config<'a>(
     ret.into_result(())
 }
 
+pub fn get_infos<'a>(
+    new_type_map: &TypeMap,
+    policies: &'a [PolicyFile],
+    classlist: &ClassList,
+    system_configs: &Option<&BTreeMap<String, &'a Argument>>,
+) -> Result<Vec<sexp::Sexp>, CascadeErrors> {
+    let ret = CascadeErrors::new();
+    // Get the policy files
+    let new_policies = get_policy_files(new_type_map, policies);
+
+    // Get the function infos
+    let mut new_func_map = get_funcs(&new_policies, new_type_map)?;
+
+    // Validate functions
+    let new_func_map_copy = new_func_map.clone(); // In order to read function info while mutating
+    validate_functions(
+        &mut new_func_map,
+        new_type_map,
+        classlist,
+        &new_func_map_copy,
+    )?;
+
+    // Get the policy rules
+    let new_policy_rules = get_policy_rules(&new_policies, new_type_map, classlist, &new_func_map)?;
+
+    let new_cil_tree = generate_sexp(
```

Comment:
Do we have to do this here?  It would be nice if generate_sexp() happened after the system loop so that we can get compile errors on all the systems before getting errors from generate_sexp().

Reply:
This comment has not yet been addressed, but I pushed the changes anyway, in case you wanted to take a look at them.
I'm running into errors with returning values referencing local variables that I haven't been able to work out, but will try again tomorrow.